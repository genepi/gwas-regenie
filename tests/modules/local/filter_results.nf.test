nextflow_process {

    name "Test Process FILTER_RESULTS"
    script "modules/local/filter_results.nf"
    process "FILTER_RESULTS"

    test("Should output only 117 filtered lines") {

        when {
            params {
                annotation_min_log10p = 1
            }
            process {
                """
                input[0] = Channel.of( ['Y1', '$baseDir/tests/input/filter_results/regenie_step2_example_Y1.regenie.gz'], ['Y2', '$baseDir/tests/input/filter_results/regenie_step2_example_Y2.regenie.gz'])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            }

        }

        test("Should output all regenie lines") {

                when {
                    params {
                        annotation_min_log10p = 0
                    }
                    process {
                        """
                        input[0] = Channel.of( ['Y1', "$baseDir/tests/input/filter_results/regenie_step2_example_Y1.regenie.gz"], ['Y2', "$baseDir/tests/input/filter_results/regenie_step2_example_Y2.regenie.gz"])
                        """
                    }
                }

                then {
                    assert process.success
                    assert snapshot(process.out).match()

                }

        }

        test("Should output an empty file") {

                when {
                    params {
                        annotation_min_log10p = 100
                    }
                    process {
                        """
                        input[0] = Channel.of( ['Y1', "$baseDir/tests/input/filter_results/regenie_step2_example_Y1.regenie.gz"], ['Y2', "$baseDir/tests/input/filter_results/regenie_step2_example_Y2.regenie.gz"])
                        """
                    }
                }

                then {
                    assert process.success
                    assert snapshot(process.out).match()
                    with(process.out){
                        with(results_filtered){
                            def lines = path(get(0).get(1)).readLinesGzip()
                            assert (lines.size() == 0)
                        }
                    }
                }

        }
}