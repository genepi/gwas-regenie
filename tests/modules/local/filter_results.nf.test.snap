{
    "Should output all regenie lines": {
        "content": [
            {
                "0": [
                    [
                        "Y1",
                        "Y1.regenie.filtered.gz:md5,d637f770b60b67e3fe3a5454dadc143b"
                    ],
                    [
                        "Y2",
                        "Y2.regenie.filtered.gz:md5,6febee3426eaca4ca8b72540ba75e633"
                    ]
                ],
                "results_filtered": [
                    [
                        "Y1",
                        "Y1.regenie.filtered.gz:md5,d637f770b60b67e3fe3a5454dadc143b"
                    ],
                    [
                        "Y2",
                        "Y2.regenie.filtered.gz:md5,6febee3426eaca4ca8b72540ba75e633"
                    ]
                ]
            }
        ],
        "timestamp": "2023-12-15T10:30:54.852916"
    },
    "Should output only 117 filtered lines": {
        "content": [
            {
                "0": [
                    [
                        "Y1",
                        "Y1.regenie.filtered.gz:md5,973ef7138c333f4c2d2787fad6c83f50"
                    ],
                    [
                        "Y2",
                        "Y2.regenie.filtered.gz:md5,d471f295b41ae96c0b094faa464c549b"
                    ]
                ],
                "results_filtered": [
                    [
                        "Y1",
                        "Y1.regenie.filtered.gz:md5,973ef7138c333f4c2d2787fad6c83f50"
                    ],
                    [
                        "Y2",
                        "Y2.regenie.filtered.gz:md5,d471f295b41ae96c0b094faa464c549b"
                    ]
                ]
            }
        ],
        "timestamp": "2023-12-15T10:16:52.606586"
    },
    "Should output an empty file": {
        "content": [
            {
                "0": [
                    [
                        "Y1",
                        "Y1.regenie.filtered.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "Y2",
                        "Y2.regenie.filtered.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results_filtered": [
                    [
                        "Y1",
                        "Y1.regenie.filtered.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "Y2",
                        "Y2.regenie.filtered.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "timestamp": "2023-12-15T10:37:15.316506"
    }
}